lib/policies/proxy/strategies/round-robin.js

```diff
@@ -0,0 +1,20 @@
+const url = require('url');
+
+module.exports = class RoundRobin {
+  constructor (proxyOptions, endpoints) {
+    this.proxyOptions = proxyOptions;
+    this.endpoints = endpoints;
+    this.endpointIndex = 0;
+    this.endpointMaxIndex = this.endpoints.length - 1;
+  }
+
+  nextTarget () {
+    const target = this.endpoints[this.endpointIndex++];
+
+    if (this.endpointIndex > this.endpointMaxIndex) {
+      this.endpointIndex = 0;
+    }
+
+    return url.parse(target);
```

Comment:
This object can be cached, so we save the parsing cost when the cycle is over.

Reply:
I've added caching for static. but this round-robin strategy will be updated later to support providers, so it can be serviceEndpoint in eg config (as it is now) as well as consul, etcd, k8s service etc. so it will need to update IP periodically in any case. 

This optimization will be done in that context